### DISCUSSION TITLE
Entailment rules reasoner

### DISCUSSION THREAD

**Discussion by prerak01** [RDFLib/rdflib#1886] [Ideas]
I have implemented a simple reasoner which follows the entailment rules as given in https://www.w3.org/TR/rdf11-mt/#entailment-rules-informative. The reasoner iterates over all the n triples in the model object and then finds entailment rule patterns in them.

**Comment by akhil821100** [comment]
Moreover, one of the entailment rules is for the transitivity of subclass. Our module supports the transitivity entailment upto any depth. We have implemented it using adjacency list representation of graph and and then applying DFS on this graph. 
Other entailments rules have been implemented by iterating over the triples of the graph and then working on the predicate.

**Comment by aucampia** [comment]
We have a reasoner here: https://github.com/RDFLib/OWL-RL
